java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageSerializer.java

```diff
@@ -266,8 +266,15 @@ public static ArrowBlock serialize(WriteChannel out, ArrowRecordBatch batch, Ipc
     long bufferLength = writeBatchBuffers(out, batch);
     Preconditions.checkArgument(bufferLength % 8 == 0, "out is not aligned");
 
+    long compLength = 0L;
+    if (batch.getBodyCompression() != null) {
+      compLength = writeCompressionBody(out, batch.getBodyCompression());
+      Preconditions.checkArgument(compLength == ArrowBodyCompression.BODY_COMPRESSION_LENGTH,
```

Comment:
Could you add a comment here on why we need to verify this value here?.  Also, it looks like align() is called in this method, so couldn't that cause these values to be unequal

Reply:
You are right. This check is removed. Thanks.